posthog/temporal/ai/sync_vectors.py

```diff
@@ -0,0 +1,270 @@
+import asyncio
+import json
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import cast
+
+import cohere
+import posthoganalytics
+import structlog
+import temporalio.activity
+import temporalio.common
+import temporalio.exceptions
+import temporalio.workflow
+from cohere.core.api_error import ApiError as CohereApiError
+from django.db.models import F, Q
+from openai import APIError as OpenAIAPIError
+
+from ee.hogai.summarizers.chains import abatch_summarize_actions
+from posthog.models import Action, FeatureFlag
+from posthog.models.ai.pg_embeddings import INSERT_BULK_PG_EMBEDDINGS_SQL
+from posthog.temporal.common.base import PostHogWorkflow
+from posthog.temporal.common.clickhouse import get_client
+from posthog.temporal.common.utils import get_scheduled_start_time
+
+logger = structlog.get_logger(__name__)
+
+
+async def _get_orgs_from_the_feature_flag() -> list[str]:
+    feature_flag = await FeatureFlag.objects.filter(key="max-rag", team_id=1).afirst()
+    if not feature_flag:
+        return []
+    payload = feature_flag.get_payload("true")
+    try:
+        orgs = json.loads(cast(str, payload))["organizations"]
+        if isinstance(orgs, list):
+            return orgs
+    except:
+        pass
+    return []
+
+
+async def get_actions_qs(start_dt: datetime, offset: int | None = None, batch_size: int | None = None):
+    orgs = await _get_orgs_from_the_feature_flag()
```

Comment:
Shall we enable for all teams in dev, for debuggability?
(is there something that would prevent this from working?)
```suggestion
async def get_actions_qs(start_dt: datetime, offset: int | None = None, batch_size: int | None = None):
   if settings.DEBUG:
        return Action.objects.all()
    orgs = await _get_orgs_from_the_feature_flag()
```